### PR TITLE
Miner config update positive tests

### PR DESCRIPTION
**Pull request by Jayashsatolia403**

### CODE REVIEW

File: tests/cli_tests/zwalletcli_miner_update_config_test.go [hunk 1]
```diff
@@ -2,22 +2,426 @@ package cli_tests
...
+			"min_stake_per_delegate",
+			"max_charge",
+			"epoch",
+			"reward_decline_rate",
+			"reward_round_frequency",
+			"num_miner_delegates_rewarded",
+			"num_sharders_rewarded",
+			"num_sharder_delegates_rewarded",
+			"cost.add_miner",
+			"cost.add_sharder",
```
Comment (Jayashsatolia403) [inline tests/cli_tests/zwalletcli_miner_update_config_test.go:112]: Add all costs

File: tests/cli_tests/zwalletcli_miner_update_config_test.go [hunk 2]
```diff
@@ -2,22 +2,426 @@ package cli_tests
...
+	// DKG - Test case for updating t_percent to a valid value
+	// 				"0.80" A mid value for t_percent
+	// DKG - Test case for updating k_percent to a valid value
+	//				"0.82" A mid value for k_percent
+	// DKG - Test case for updating x_percent to a valid value
+	//				"0.85" A mid value for x_percent
+
+	t.RunSequentially("successful update of config to mid-level allowed value", func(t *test.SystemTest) {
+
+		keys := []string{"reward_rate", "block_reward", "share_ratio", "reward_decline_rate", "t_percent", "k_percent", "x_percent"}
```
Comment (Jayashsatolia403) [inline tests/cli_tests/zwalletcli_miner_update_config_test.go:225]: Remove t_percent, k_percent and x_percent

File: tests/cli_tests/zwalletcli_miner_update_config_test.go [hunk 3]
```diff
@@ -2,22 +2,426 @@ package cli_tests
...
+			"max_n",
+			"min_n",
+			"max_s",
+			"min_s",
+			"max_delegates",
+			"reward_rate",
+			"block_reward",
+			"share_ratio",
+			"reward_decline_rate",
+			"t_percent",
```
Comment (Jayashsatolia403) [inline tests/cli_tests/zwalletcli_miner_update_config_test.go:98]: Remove t_percent, k_percent and x_percent

File: tests/cli_tests/zwalletcli_miner_update_config_test.go [hunk 4]
```diff
@@ -2,22 +2,426 @@ package cli_tests
...
+	// 				"1"
+	// Share Ratio - Test case for updating share_ratio
+	//				"1" // A value of 1 for the exclusive range
+	// Reward Decline Rate - Test case for updating reward_decline_rate to the minimum allowed value
+	//				"1" // A value of 1 for the exclusive range
+
+
+	t.RunSequentially("unsuccessful update of config to out of bounds value", func(t *test.SystemTest) {
+
+		keys := []string{"reward_rate", "block_reward", "share_ratio", "reward_decline_rate"}
```
Comment (Jayashsatolia403) [inline tests/cli_tests/zwalletcli_miner_update_config_test.go:409]: Skip this test with issue link in comment : https://github.com/0chain/0chain/issues/3168.

**Review by Jayashsatolia403 — COMMENTED** [review]
Fix lint errors.

File: tests/cli_tests/zwalletcli_miner_update_config_test.go (hunk 1 above)
Comment (CristalWilsonLobo) [reply tests/cli_tests/zwalletcli_miner_update_config_test.go:109]: Added all costs except the below as it's not found in the config settings 

cost.contributeMpk
cost.shareSignsOrShares
cost.payFees
cost.feesPaid
cost.mintedTokens
cost.addToDelegatePool
cost.deleteFromDelegatePool
cost.collect_reward

File: tests/cli_tests/zwalletcli_miner_update_config_test.go (hunk 2 above)
Comment (CristalWilsonLobo) [reply tests/cli_tests/zwalletcli_miner_update_config_test.go]: Removed from all test suites

File: tests/cli_tests/zwalletcli_miner_update_config_test.go (hunk 3 above)
Comment (CristalWilsonLobo) [reply tests/cli_tests/zwalletcli_miner_update_config_test.go]: Removed from all test suites

File: tests/cli_tests/zwalletcli_miner_update_config_test.go (hunk 4 above)
Comment (CristalWilsonLobo) [reply tests/cli_tests/zwalletcli_miner_update_config_test.go]: Skipped test, added issue link in comments

File: tests/cli_tests/zwalletcli_miner_update_config_test.go (hunk 1 above)
Comment (Jayashsatolia403) [reply tests/cli_tests/zwalletcli_miner_update_config_test.go:110]: 0chain is reading it all small latters. Check this for example : `SettingName[CostContributeMpk] = strings.ToLower("cost.contributeMpk")`

**Pull request merged**